mobility_games/games/semantic_waypoints.py

```diff
@@ -0,0 +1,292 @@
+        tag_name = raw_input("Name the AR Tag: "); 
+        confirm_msg = "Labled " + tag_name
+        self.engine.say(confirm_msg)
+        self.tag_id_to_name[tag_id] = tag_name
+        self.tag_name_to_id[tag_name] = tag_id
+        print self.tag_id_to_name
+
+    # Gives option to users to leave a message at an AR tag with tag_id
+    def collect_message(self, tag_id):
+        prompt = "Would you like to leave a message at " + self.tag_id_to_name[tag_id] + "? [y/n] "
+        self.engine.say(prompt)
+        message_prompt = raw_input(prompt)
+        if message_prompt == 'y':
+            self.engine.say("Leve a message")
+            message = raw_input("Leave a message: ")
+            if tag_id in self.tag_messages:
+                self.tag_messages[tag_id].append(message)
+            else:
+                messages = []
+                messages.append(message)
+                self.tag_messages[tag_id] = messages
+
+    # Displays all messages at AR Tag with tag_id
+    def display_messages(self, tag_id):
+        if tag_id in self.tag_messages:
+            print "Someone has left you a message:"
+            self.engine.say('Somone has left you a message')
+            for message in self.tag_messages[tag_id]:
+                print message + '\n'
+                self.engine.say(message)
+
+    def update_destination_pose(self):
+        if self.visited_tags_calibrate[0] == self.tag_id:
+            ARFrame = 'AR'
+        else:
+            ARFrame = "AR_" + str(self.tag_id);
+
+        if (self.listener.frameExists("odom")
+                and self.listener.frameExists(ARFrame)):
+
+            try:
+                t = self.listener.getLatestCommonTime("odom", ARFrame)
+
+                # get transform to make tag_frame (parent) & odom (child)
+                trans, rot = self.listener.lookupTransform(
+                        "odom", ARFrame, t)
+
+                self.translations = trans
+                # self.rotations = rot
+
+            except (tf.ExtrapolationException,
+                    tf.LookupException,
+                    tf.ConnectivityException) as e:
+                print e
+
+
+    # Starts new game 
+    #   - Prompt user to enter destination
+    #   - Restart timer
+    #   - End game by typing 'done'
+    def start_new_game(self):
+        self.engine.say("What would you like to find?")
+        search_tag_name = raw_input("What would you like to find? ")
+        if search_tag_name in self.tag_name_to_id:
+            confirm_msg = "Looking for " + search_tag_name
+            self.engine.say(confirm_msg)
+            self.tag_id = self.tag_name_to_id[search_tag_name]
+            
+            self.update_destination_pose()
+           
+            self.start_run = True
+            self.start_time = time.time()
+        elif search_tag_name == 'done':
+            print "Goodbye!"
+        else:
+            err_msg =  '"' + search_tag_name + '" does not exisit. Try again!'
```

Comment:
Typo on "exist." Also, might be cleaner as:

`err_msg = '"%s" does not exist. Try again!' % search_tag_name`